common/scala/src/main/scala/whisk/core/containerpool/logging/SplunkLogStore.scala

```diff
@@ -0,0 +1,158 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.containerpool.logging
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+import akka.http.scaladsl.client.RequestBuilding.Post
+import akka.http.scaladsl.model.FormData
+import akka.http.scaladsl.model.HttpRequest
+import akka.http.scaladsl.model.HttpResponse
+import akka.http.scaladsl.model.headers.Authorization
+import akka.http.scaladsl.model.headers.BasicHttpCredentials
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.ActorMaterializer
+import akka.stream.OverflowStrategy
+import akka.stream.QueueOfferResult
+import akka.stream.scaladsl.Flow
+import akka.stream.scaladsl.Keep
+import akka.stream.scaladsl.Sink
+import akka.stream.scaladsl.Source
+import com.typesafe.sslconfig.akka.AkkaSSLConfig
+import scala.concurrent.Future
+import scala.concurrent.Promise
+import scala.util.Failure
+import scala.util.Success
+import scala.util.Try
+import spray.json.DefaultJsonProtocol._
+import spray.json.JsArray
+import spray.json._
+import whisk.core.entity.ActivationLogs
+import whisk.core.entity.WhiskActivation
+
+/**
+ * A Splunk based impl of LogDriverLogStore. Logs are routed to splunk via docker log driver, and retrieved via Splunk REST API
+ * @param actorSystem
+ * @param httpFlow Optional Flow to use for HttpRequest handling (to enable stream based tests)
+ */
+class SplunkLogStore(
+  actorSystem: ActorSystem,
+  httpFlow: Option[Flow[(HttpRequest, Promise[HttpResponse]), (Try[HttpResponse], Promise[HttpResponse]), Any]] = None)
+    extends LogDriverLogStore(actorSystem) {
+  implicit val as = actorSystem
+  implicit val ec = as.dispatcher
+  implicit val materializer = ActorMaterializer()
+
+  val splunkHost = config.getString("whisk.logstore.splunk.host")
+  val splunkPort = config.getInt("whisk.logstore.splunk.port")
+  private val splunkApi = "/services/search/jobs" //see http://docs.splunk.com/Documentation/Splunk/6.6.3/RESTREF/RESTsearch#search.2Fjobs
+  private val splunkUser = config.getString("whisk.logstore.splunk.user")
+  private val splunkPass = config.getString("whisk.logstore.splunk.password")
+  private val splunkIndex = config.getString("whisk.logstore.splunk.index")
+  private val logMessageFieldName = config.getString("whisk.logstore.splunk.log-message-field")
+  private val activationIdFieldName = config.getString("whisk.logstore.splunk.activation-id-field")
+  private val disableSNI = config.getBoolean("whisk.logstore.splunk.disableSNI")
```

Comment:
I think it would be valuable to use #2576 (pureconfig) to parse all of these into a case-class. Will be much less verbose.